Feature/te 260 test stop stepwise in vm

kpanier opened the pull request:
Changing property rules for headless runner. 
Command line arguments now overrides workspace settings in the headless runner mode.
The debug port is mapped in the vagrant file of the tests in the test-editor-test repository.

kpanier committed:
TE-260 Enable test pause in test execution environment

- Add support to prefer vriables from system over local preferences

https://testeditor.atlassian.net/browse/TE-260

kpanier committed:
TE-260 Enable Test pause in testexecution 

Task-Url: https://testeditor.atlassian.net/browse/TE-260

franzbecker commented on core/org.testeditor.core/src/main/java/org/testeditor/core/headless/HeadlessTestRunnerApplication.java:
```diff
@@ -240,18 +239,8 @@ public TestStructure getTestStructureToExecute(String[] args)
-		String waittime = System.getProperty(TestEditorGlobalConstans.DEFINE_WAITS_AFTER_TEST_STEP);
-		String browserPath = System.getProperty(TestEditorGlobalConstans.PATH_BROWSER);
-		testEditorConfigService.exportGlobalVariablesToSystemProperties();
+		testEditorConfigService.exportGlobalVariablesToSystemProperties(false);
```
I think the readability would improve by refactoring the variable name to `configService`, same for `TestEditorConfigurationServiceImplIntTest`

franzbecker commented on core/org.testeditor.core/src/main/java/org/testeditor/core/services/impl/TestEditorConfigurationServiceImpl.java:
```diff
@@ -64,12 +64,18 @@ protected String fixWhiteSpaceOnPreferenceString(String replaceStr) {
 
 	@Override
-	public void exportGlobalVariablesToSystemProperties() throws BackingStoreException {
+	public void exportGlobalVariablesToSystemProperties(boolean overRide) throws BackingStoreException {
```
I don't see a reason for camel-case here, `override` would be just fine.

franzbecker commented:
It would be nice if we had a unit test for `TestEditorConfigService.exportGlobalVariablesToSystemProperties` for the two cases of `override` as well.

kpanier committed:
TE-260 Enable Test pause in testexecution 

- improved readability

Task-Url: https://testeditor.atlassian.net/browse/TE-260

kpanier committed:
TE-260 Enable Test pause in testexecution 

- improved readability

Task-Url: https://testeditor.atlassian.net/browse/TE-260

kpanier committed:
TE-260 Enable Test pause in testexecution 

- adding integration tests

Task-Url: https://testeditor.atlassian.net/browse/TE-260

kpanier commented:
Fixing:
- the method parameter
- rename service variable
- adding integration tests
